plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -545,4 +558,12 @@ public void testExecuteProcedure()
             assertUpdate("DROP TABLE IF EXISTS " + schemaTableName);
         }
     }
+
+    @Test
+    @Override
+    public void testDeleteWithSubquery()
+    {
+        // TODO (https://github.com/trinodb/trino/issues/170037) Remove once the issue is solved
+        assertThatThrownBy(super::testDeleteWithSubquery).hasMessageContaining("Unexpected Join over for-update table scan");
```

Comment:
Is this a regression? Why it fails and how it was working before?

Reply:
> Why it fails and how it was working before?

It fails because of the optimizer `PushJoinIntoTableScan` see https://github.com/trinodb/trino/pull/24447 . 
`Ignite` connector can hit this because it's can pushdown the expression `LIKE 'A%'` such expression, see `connectorExpressionRewriter` in `IgniteClient`. 

Postgresql varchar by default not support such expression, it controls varchar type pushdown in `PostgresqlClient#POSTGRESQL_STRING_COLLATION_AWARE_PUSHDOWN`, if we enable this experimental pushdown for varchar with collate will hit the same problem in Postgresql.